### DISCUSSION TITLE
Useful Ubuntu 22.04/24.04 VM Commands

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#2072] [Announcements]
## Setting up Cloud-Init
![Screenshot 2023-11-18 3 56 52 PM](https://github.com/tteck/Proxmox/assets/21342670/9f7d7278-edb9-4d31-9b8d-ec58c92c30b2)

1) Set user to root
2) Give root user a password
3) Set to no (not enough storage, yet)
4) Set network settings
5) Regenerate Image

**Now you can start the VM**

### Select xterm.js under the Console pull down for copy/paste functions
![Screenshot 2023-11-16 9 48 54 AM](https://github.com/tteck/Proxmox/assets/21342670/2a01de8c-4f08-41ef-a815-c2a807a3dfba)

### Hardware > Hard Disk (scsi0) > Disk Action > Resize
![Screenshot 2023-11-16 11 56 40 AM](https://github.com/tteck/Proxmox/assets/21342670/4b9d02aa-e5a6-4270-a442-3340fafca4ed)

## Expand VM Disk using parted (/dev/sda1)

`parted /dev/sda`
`resizepart 1`
Fix/Ignore? `Fix`
Partition number? `1`
Yes/No? `Yes`
End?  [2146MB]? `-0`
(parted) `quit`
**reboot**

## Get SSH Going
```
sed -i -e 's/#PermitRootLogin prohibit-password/PermitRootLogin yes/g' -e 's/^PasswordAuthentication.*/PasswordAuthentication yes/' /etc/ssh/sshd_config
rm /etc/ssh/sshd_config.d/60-cloudimg-settings.conf
systemctl restart sshd
```

## Add Guest Agent
```
apt-get update && apt-get -y upgrade
apt-get install -y qemu-guest-agent
```
**reboot**
## Install Docker
```
sh <(curl -sSL https://get.docker.com)
```
## Install Docker Compose
```
LATEST=$(curl -sL https://api.github.com/repos/docker/compose/releases/latest | grep '"tag_name":' | cut -d'"' -f4)
DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
mkdir -p $DOCKER_CONFIG/cli-plugins
curl -sSL https://github.com/docker/compose/releases/download/$LATEST/docker-compose-linux-x86_64 -o ~/.docker/cli-plugins/docker-compose
chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
docker compose version
```
## Add Dockge
* Install Docker Compose first
```
mkdir -p /opt/{dockge,stacks}
wget -q -O /opt/dockge/compose.yaml https://raw.githubusercontent.com/louislam/dockge/master/compose.yaml
cd /opt/dockge
docker compose up -d
```
Dockge GUI, **https:// IP:5001**
## Add Portainer
* Docker run
```
docker volume create portainer_data
docker run -d \
  -p 8000:8000 \
  -p 9443:9443 \
  --name=portainer \
  --restart=always \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v portainer_data:/data \
  portainer/portainer-ce:latest
```
Portainer GUI, **https:// IP:9443**

**Comment by AOskam** [comment]
Hi! using the script, ubuntu installs perfectly, however standard login ubuntu/ubuntu is not accepted. Proxmox 8.

**Reply by tteck** [reply]
Did you follow "Setting up Cloud-Init"?

**Reply by AOskam** [reply]
Reading is difficult :). Thanks

**Comment by chriss-89** [comment]
Hi and thanks for the scripts. They are such a relief. 
However, I have the following problem with the Ubuntu VM:
I have run the "Get SSH Going" section, but I cannot connect with PuTTY or Termux via ssh (connecting with root and password). I get the following message: No auth methods could be used.
I do not have this problem with the Debian 12 VM.
Does anything else need to be adjusted in the sshd config or how can I fix the error @tteck ?

**Reply by Necrul** [reply]
Did you ever figure this out? Running into the same issue here. @tteck

**Reply by iz8mbw** [reply]
just remove or rename the file `60-cloudimg-settings.conf` in `/etc/ssh/sshd_config.d`

**Comment by elishaip** [comment]
I ran the Ubuntu VM but see no GUI. How do we proceed from here? I am completely new to this.

![image](https://github.com/tteck/Proxmox/assets/89676547/823a458d-6b90-4fd3-9bab-301daeb96c5c)

**Reply by thiagogpa** [reply]
This is Ubuntu server, there's no GUI

**Comment by JZambranoIT** [comment]
It's meant for servers so you won't get a gui.

If you want a desktop environment create a vm through the proxmox gui using
an ubuntu desktop iso.

**Reply by elishaip** [reply]
I understood. Thanks.

**Reply by alienatedsec** [reply]
or install gui :) `sudo apt install ubuntu-desktop` and use the console

![image](https://github.com/user-attachments/assets/bc0d4fc5-1e22-470c-99f9-6be7c13fc2be)

**Reply by Rockel83** [reply]
Oh... I also was not aware that this was only the server... I also thought it was the GUI-version, while I can not finding anything back from it on the script page.
I was actually looking putting a light OS (with GUI) on my Proxmox. So installing the GUI after would be possible. But are the script standards still sufficient then?

"Default Settings: 2GB RAM - 2GB Storage - 2vCPU"

**Comment by sugar012** [comment]
it's time for 24.04!

**Comment by netxer** [comment]
Hey I have couple of questions regarding this way of generationg VMs,

1. Why not implament  inside the sciprt to use cloud init rather doing it manuly? 
2. Is there's a documents on ubuntu image where I can find which pakcages are already installed?

**Reply by tteck** [reply]
1) The script sets it as `cloudinit`
https://github.com/tteck/Proxmox/blob/cae357cc262d2b26a39a81249faa469225c43188/vm/ubuntu2204-vm.sh#L411

2) 🤷🏻‍♂️

**Reply by netxer** [reply]
Just gonna post here the issue I encountered after I installed that vm (it might help someone who got stuck on different things related to this img).
I had an issue with mounting a cifs shares into this vm, as cifs module isn't present inside the kernal package of this img, so to fix that issue is I found out you had to install extra modules by doing "sudo apt install linux-modules-extra-6.8.0-31-generic" which fixed my issue after a reboot.

**Comment by project-blaze** [comment]
I am trying to mount the smb share from truenas using latest ubuntu 24.04, but it doesn't work using the configuration below. It always fails. I have already installed cifs-utils and tried the following entry in fstab. I have even tried including the username and password directly.
//192.168.1.21/data /mnt/truenas/data cifs credentials=/etc/samba/.smbcreds 0 0
#//192.168.1.21/data /mnt/truenas/data cifs vers=3.0,credentials=/etc/samba/.smbcreds,uid=1000,gid=1000,defaults 0 0

sudo mount -a
mount error: cifs filesystem not supported by the system
mount error(19): No such device
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs) and kernel log messages (dmesg)

Update:
Sorry, I didn't see the solution above. This command solved the issue by the time of posting.

'''sudo apt install linux-modules-extra-6.8.0-31-generic'''

**Comment by trungsky** [comment]
Get SSH Going in Ubuntu 24.04

`
systemctl restart ssh
`

**Reply by mysteriousHerb** [reply]
is it possible to include this update in the document please :)

**Comment by weir1** [comment]
I am facing issue, VM is restarting after every few mins

**Comment by SebastianS1986** [comment]
Hello, I'm new here - please don't stone me for asking! What is the username and password? If I just enter "root" and no password - it tells me wrong login details

**Reply by That-Dude** [reply]
In Proxmox click the 'Clout-Init' section for the VM, set your password, then 'regenerate image'

**Reply by SebastianS1986** [reply]
Thank you very much - I can't find it! Can you tell me where I can find it better?

**Reply by SebastianS1986** [reply]
it worked! thank you very much!

**Comment by eagle275** [comment]
Help!? - As I want to follow a video to setup openstack using ubuntu server I'm very grateful for the 22.04 LTS image.
I've only recently setup proxmox pve and learn the ropes. I found that all(?) of your offered images share the same ID (100) - is it easy to change that or can I install multiple instances with the same ID? Sorry if my question is dumb or easily answered elsewhere

**Comment by ForTheRush** [comment]
any wat to remove cloud-init? apt remove didnt seem to work and i very much would not like to use it
